native/core/src/execution/datafusion/planner.rs

```diff
@@ -1277,8 +1277,6 @@ impl PhysicalPlanner {
             }
             AggExprStruct::Min(expr) => {
                 let child = self.create_expr(expr.child.as_ref().unwrap(), schema.clone())?;
-                let datatype = to_arrow_datatype(expr.datatype.as_ref().unwrap());
-                let child = Arc::new(CastExpr::new(child, datatype.clone(), None));
```

Comment:
This is not related to window function `sum` support. 
`CastExpr` is not needed for min and max, so remove it.